Update Cloudflare deploy/integration docs to mention Workers Assets (#11344)
withastro/docs: src/content/docs/en/guides/deploy/cloudflare.mdx

sidebar:
  label: Cloudflare
type: deploy
i18nReady: true
---
import ReadMore from '~/components/ReadMore.astro';
import { Steps } from '@astrojs/starlight/components';
import StaticSsrTabs from '~/components/tabs/StaticSsrTabs.astro';

You can deploy full-stack applications, including front-end static assets and back-end APIs, as well as SSR sites, to both [Cloudflare Workers](https://developers.cloudflare.com/workers/static-assets/) and [Cloudflare Pages](https://pages.cloudflare.com/). 

This guide includes:

- [How to deploy to Cloudflare Workers](#cloudflare-workers)
- [How to deploy to Cloudflare Pages](#cloudflare-pages)

<ReadMore>Read more about [using the Cloudflare runtime](/en/guides/integrations-guide/cloudflare/) in your Astro project.</ReadMore>
## Prerequisites

To get started, you will need:

- A Cloudflare account. If you don’t already have one, you can create a free Cloudflare account during the process.

## Cloudflare Workers

### How to deploy with Wrangler

<Steps>
1. Install [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/).
    ```bash
      npm install wrangler@latest --save-dev
    ```

> 2. If your site uses SSR, install the `@astrojs/cloudflare` adapter.

Reviewer: ```suggestion
2. If your site uses on demand rendering, install the `@astrojs/cloudflare` adapter.
```

    This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step.
      ```bash
        npx astro add cloudflare
      ```

>     <ReadMore>Read more about [SSR in Astro](/en/guides/on-demand-rendering/).</ReadMore>

Reviewer: ```suggestion
    <ReadMore>Read more about [on demand rendering (also known as SSR) in Astro](/en/guides/on-demand-rendering/).</ReadMore>
```

> 2. Create a [Wrangler configuration file](https://developers.cloudflare.com/workers/wrangler/configuration/)

Reviewer: ```suggestion
2. Create a [Wrangler configuration file](https://developers.cloudflare.com/workers/wrangler/configuration/).
```

    ```

4. Deploy using `npx wrangler deploy`.
    ```bash
      npx astro build && npx wrangler deploy
    ```
</Steps>

After your assets are uploaded, Wrangler will give you a preview URL to inspect your site.

<ReadMore>Read more about using [Cloudflare runtime APIs](/en/guides/integrations-guide/cloudflare/) such as bindings.</ReadMore>

### How to deploy with CI/CD

You can also use a CI/CD system such as [Workers Builds (BETA)](https://developers.cloudflare.com/workers/ci-cd/builds/) to automatically build and deploy your site on push. 

If you're using Workers Builds:

<Steps>
1. Follow Steps 1-3 from the Wrangler section above.
2. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/) and navigate to `Workers & Pages`. Select `Create`.
3. Under `Import a repository`, select a Git account and then the repository containing your Astro project.
4. Configure your project with:
    - Build command: `npx astro build`
    - Deploy command: `npx wrangler deploy`
5. Click `Save and Deploy`. You can now preview your Worker at its provided `workers.dev` subdomain.
</Steps>

## Cloudflare Pages

### How to deploy with Wrangler

<Steps>

> 1. Install [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/).

Reviewer: ```suggestion
1. Install [Wrangler CLI].(https://developers.cloudflare.com/workers/wrangler/get-started/).
```

Reviewer 2: @sarah11918 is this a mistake, or some Markdown syntax I've not encountered?

Author: definitely an accident 😅

    ```bash
      npm install wrangler@latest --save-dev
    ```

> 2. If your site uses SSR, install the `@astrojs/cloudflare` adapter 

Reviewer: ```suggestion
2. If your site uses on demand rendering, install the `@astrojs/cloudflare` adapter. 
```

    This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step.
      ```bash
        npx astro add cloudflare
      ```

>     <ReadMore>Read more about [SSR in Astro](/en/guides/on-demand-rendering/).</ReadMore>

Reviewer: ```suggestion
    <ReadMore>Read more about [on demand rendering in Astro](/en/guides/on-demand-rendering/).</ReadMore>
```

3. Preview your project locally with Wrangler.
    ```bash
      npx astro build && npx wrangler pages dev ./dist
    ```

4. Deploy using `npx wrangler deploy`.
    ```bash
      npx astro build && npx wrangler pages deploy ./dist
    ```
</Steps>

After your assets are uploaded, Wrangler will give you a preview URL to inspect your site.

### How to deploy a site with Git

<Steps>

> 1. Push your code to your git repository (GitHub, GitLab).

Reviewer: ```suggestion
1. Push your code to your git repository (e.g. GitHub, GitLab).
```

> 2. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/) and navigate to `Workers & Pages`. Select `Create` and then select the `Pages` tab. Connect your git repository.

Reviewer 3: I thought that we want a generic instruction instead of a detailed one? As mentioned here: https://github.com/withastro/docs/pull/10790#issuecomment-2630828967

Reviewer: @vrabe Ooh, you're right, we missed this one! Would you like to make a PR to update with your suggestion?

---
import ReadMore from '~/components/ReadMore.astro';
import { Steps } from '@astrojs/starlight/components';

> import StaticSsrTabs from '~/components/tabs/StaticSsrTabs.astro';

Reviewer 4: I was not aware this component existed either :smile:

import { Steps } from '@astrojs/starlight/components';

You can deploy full-stack applications, including front-end static assets and back-end APIs, as well as SSR sites, to both [Cloudflare Workers](https://developers.cloudflare.com/workers/static-assets/) and [Cloudflare Pages](https://pages.cloudflare.com/). 

This guide includes:

- [How to deploy to Cloudflare Workers](#cloudflare-workers)
- [How to deploy to Cloudflare Pages](#cloudflare-pages)

## Prerequisites

To get started, you will need:

- A Cloudflare account. If you don’t already have one, you can create a free Cloudflare account during the process.

## Cloudflare Workers

### How to deploy with Wrangler

<Steps>
1. Install [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/).
    ```bash
      npm install wrangler@latest --save-dev
    ```

2. If your site uses SSR, install the `@astrojs/cloudflare` adapter 
    This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step.
      ```bash
        npx astro add cloudflare
      ```
    <ReadMore>Read more about [SSR in Astro](/en/guides/on-demand-rendering/).</ReadMore>

> 2. Create a [Wrangler configuration file](https://developers.cloudflare.com/workers/wrangler/configuration/)

Reviewer 4: Would it be useful to use tabs for SSG/SSR wrangler configs?

Author: oh neat i had no idea that component existed :)

    If your Astro project is purely static, you can use the following configuration:

>     ```json title="wrangler.jsonc"

Reviewer 2: ```suggestion
    ```jsonc title="wrangler.jsonc"
```

      {
        "name": "my-astro-app",
        // Update to today's date
        "compatibility_date": "2025-03-25",
        "assets": {
          "directory": "./dist"
        }
      }
    ```

    If your Astro project uses SSR, you can use the following configuration:

>     ```json title="wrangler.jsonc"

Reviewer 2: ```suggestion
    ```jsonc title="wrangler.jsonc"
```
